src/jdk.compiler/share/classes/com/sun/tools/javac/file/JavacFileManager.java

```diff
@@ -957,17 +957,16 @@ private JavaFileObject getFileForOutput(Location location,
     }
 
     @Override @DefinedBy(Api.COMPILER)
-    public Iterable<? extends JavaFileObject> getJavaFileObjectsFromPaths(
-        Collection<? extends Path> paths)
-    {
+    public Iterable<? extends JavaFileObject> getJavaFileObjectsFromPaths(Collection<? extends Path> paths) {
         ArrayList<PathFileObject> result;
-        if (paths instanceof Collection<?>)
-            result = new ArrayList<>(((Collection<?>)paths).size());
-        else
+        if (paths != null) {
+            result = new ArrayList<>(paths.size());
```

Comment:
This seems odd in the original code - I'd rather have somebody versed in the API double check this - @jonathan-gibbons  ? Anyway - this doesn't seem to be related with using pattern matching, so perhaps it's preferrable to just move it out of this PR.

Reply:
OK